src/contracts/core/StrategyManager.sol

```diff
@@ -186,129 +183,38 @@ contract StrategyManager is
-     * popping off the last entry in `stakerStrategyList`. The simplest way to calculate the correct `strategyIndexes` to input
-     * is to order the strategies *for which `msg.sender` is withdrawing 100% of their shares* from highest index in
-     * `stakerStrategyList` to lowest index
-     */
-    function queueWithdrawal(
-        uint256[] calldata strategyIndexes,
-        IStrategy[] calldata strategies,
-        uint256[] calldata shares,
-        address withdrawer
-    ) external onlyWhenNotPaused(PAUSED_WITHDRAWALS) onlyNotFrozen(msg.sender) nonReentrant returns (bytes32) {
-        bytes32 queuedWithdrawal = _queueWithdrawal(msg.sender, strategyIndexes, strategies, shares, withdrawer);
-        delegation.decreaseDelegatedShares(msg.sender, strategies, shares);
-        return queuedWithdrawal;
+    function addShares(
+        address grantee,
+        IStrategy strategy,
+        uint256 shares
+    ) external onlyDelegationManager {
+        _addShares(grantee, strategy, shares);
     }
 
-    /**
-     * @notice Used to complete the specified `queuedWithdrawal`. The function caller must match `queuedWithdrawal.withdrawer`
-     * @param queuedWithdrawal The QueuedWithdrawal to complete.
-     * @param tokens Array in which the i-th entry specifies the `token` input to the 'withdraw' function of the i-th Strategy in the `strategies` array
-     * of the `queuedWithdrawal`. This input can be provided with zero length if `receiveAsTokens` is set to 'false' (since in that case, this input will be unused)
-     * @param middlewareTimesIndex is the index in the operator that the staker who triggered the withdrawal was delegated to's middleware times array
-     * @param receiveAsTokens If true, the shares specified in the queued withdrawal will be withdrawn from the specified strategies themselves
-     * and sent to the caller, through calls to `queuedWithdrawal.strategies[i].withdraw`. If false, then the shares in the specified strategies
-     * will simply be transferred to the caller directly.
-     * @dev middlewareTimesIndex should be calculated off chain before calling this function by finding the first index that satisfies `slasher.canWithdraw`
-     */
-    function completeQueuedWithdrawal(
-        QueuedWithdrawal calldata queuedWithdrawal,
-        IERC20[] calldata tokens,
-        uint256 middlewareTimesIndex,
-        bool receiveAsTokens
-    )
-        external
-        onlyWhenNotPaused(PAUSED_WITHDRAWALS)
-        // check that the address that the staker *was delegated to* – at the time that they queued the withdrawal – is not frozen
-        nonReentrant
-    {
-        _completeQueuedWithdrawal(queuedWithdrawal, tokens, middlewareTimesIndex, receiveAsTokens);
+    function withdrawSharesAsTokens(
+        address destination,
+        IStrategy strategy,
+        uint256 shares,
+        IERC20 token
```

Comment:
token, shares is a nicer order imo

Reply:
we use 'staker, Strategy, shares' in multiple other functions, so I feel like this ordering is at least pretty consistent.

Reply:
it's consistent with the way the EPmgr is called, and makes it clear that `tokens` is this extra thing the Smgr needs. I like this order.